torchvision/datasets/kitti.py

```diff
@@ -0,0 +1,166 @@
+
+    def __init__(
+        self,
+        root: str,
+        split: str = None,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        transforms: Optional[Callable] = None,
+        download: bool = False,
+    ):
+        super().__init__(
+            root,
+            transform=transform,
+            target_transform=target_transform,
+            transforms=transforms,
+        )
+        self.images = []
+        self.targets = []
+        self.root = root
+        self.split = split
+
+        if download:
+            self.download()
+        if not self._check_exists():
+            raise RuntimeError(
+                "Dataset not found. You may use download=True to download it."
+            )
+
+        location = "testing" if self.split == "test" else "training"
+        image_dir = os.path.join(self.raw_folder, location, self.image_dir_name)
+        if location == "training":
+            labels_dir = os.path.join(self.raw_folder, location, self.labels_dir_name)
+        for img_file in os.listdir(image_dir):
+            self.images.append(os.path.join(image_dir, img_file))
+            if location == "training":
+                self.targets.append(
+                    os.path.join(labels_dir, f"{img_file.split('.')[0]}.txt")
+                )
+
+    def __getitem__(self, index: int) -> Tuple[Any, Any]:
+        """Get item at a given index.
+        Args:
+            index (int): Index
+        Returns:
+            tuple: (image, target), where
+            target is a list of dictionaries with the following keys:
+                type: str
+                truncated: float
+                occluded: int
+                alpha: float
+                bbox: float[4]
+                dimensions: float[3]
+                locations: float[3]
+                rotation_y: float
+        """
+        image = Image.open(self.images[index])
+        target = None if self.split == "test" else self._parse_target(index)
+        if self.transforms:
+            image, target = self.transforms(image, target)
+        return image, target
+
+    def _parse_target(self, index: int) -> List:
+        target = []
+        with open(self.targets[index]) as inp:
+            content = csv.reader(inp, delimiter=" ")
+            for line in content:
+                target.append({
+                    "type": line[0],
+                    "truncated": float(line[1]),
+                    "occluded": int(line[2]),
+                    "alpha": float(line[3]),
+                    "bbox": [float(x) for x in line[4:8]],
+                    "dimensions": [float(x) for x in line[8:11]],
+                    "location": [float(x) for x in line[11:14]],
+                    "rotation_y": float(line[14]),
+                })
+        return target
+
+    def __len__(self) -> int:
+        return len(self.images)
+
+    @property
+    def raw_folder(self) -> str:
```

Comment:
Unless this is expected to be public, I'd suggest to make it private: `_raw_folder`. By making it private we're protecting ourselves against the impossiblity of making BC breaking changes, as it happened in https://github.com/pytorch/vision/pull/3611#issuecomment-810352148

Reply:
We may want to update `mnint.py` as well(I can create a separate PR for that).